java/arcs/core/data/proto/TypeProtoDecoders.kt

```diff
@@ -30,7 +30,15 @@ fun PrimitiveTypeProto.decode() = when (this) {
 }
 
 /** Converts a [PrimitiveTypeProto] protobuf instance into a Kotlin [FieldType] instance. */
-fun PrimitiveTypeProto.decodeAsFieldType(): FieldType.Primitive = FieldType.Primitive(decode())
+fun PrimitiveTypeProto.decodeAsFieldType() = FieldType.Primitive(decode())
+
+/** Converts a [ReferenceTypeProto] protobuf instance into a Kotlin [FieldType] instance. */
+fun ReferenceTypeProto.decodeAsFieldType() = FieldType.EntityRef(decode().entitySchema!!.hash)
```

Comment:
I found it difficult to wrap the `.hash` accessor with `requireNotNull`. Any ideas for a better way?

Reply:
Doesn't `requireNotNull(decode().entitySchema).hash` work? 

I think it might good to be a explicit here:
```
fun ReferenceTypeProto.decodeAsFieldType(): FieldType.EntityRef {
   val entitySchema = requireNotNull(decode().entitySchema) { 
        "Field that is a reference to an non-entity type is not possible." 
   } 
   return FieldType.EntityRef(entitySchema.hash)
}
```

Reply:
I like that err msging.